Exercise from Think Stats, 2nd Edition (thinkstats2.com)<br>
Allen Downey

In [1]:
%matplotlib inline

Using the data from the BRFSS, compute the linear least squares fit for log(weight) versus height. How would you best present the estimated parameters for a model like this where one of the variables is log-transformed? If you were trying to guess someone’s weight, how much would it help to know their height? 

In [2]:
import brfss
from matplotlib import pyplot as plt
import seaborn as sns
import math

In [3]:
df = brfss.ReadBrfss()

In [4]:
df = df.dropna(subset=['wtkg2', 'htm3'])

In [25]:
weight = df.wtkg2
log_weight = weight.apply(math.log)
height = df.htm3

0         157
1         163
3         170
4         185
5         183
6         157
7         178
8         178
9         173
10        170
11        168
12        175
13        183
14        175
15        157
16        168
17        155
18        160
19        165
20        157
21        155
22        165
23        183
24        160
25        157
26        173
27        157
28        157
29        165
30        165
         ... 
414476    170
414477    180
414478    163
414479    160
414480    178
414481    178
414482    157
414483    160
414484    170
414485    157
414486    168
414488    175
414489    163
414490    170
414492    157
414493    170
414494    152
414495    165
414496    157
414497    163
414498    173
414499    173
414500    168
414501    175
414503    168
414504    191
414505    170
414506    178
414507    157
414508    178
Name: htm3, dtype: float64

In [24]:
from thinkstats2 import LeastSquares
inter, slope = LeastSquares(height, log_weight)
print "Slope: %.3f, Intercept: %.2f" % (slope, inter)

Slope: 0.012, Intercept: 2.29


In [9]:
from math import exp
predictions = (slope * height + inter).apply(exp)

In [22]:
from sklearn.metrics import mean_squared_error, r2_score
from math import sqrt
from numpy import std

rms = sqrt(mean_squared_error(weight, predictions))
print "RME Using model: %.1f" % rms
print "RME Without model: %.1f" % std(weight)
print "Percentage improvement: %i%%" % (100 * (std(weight) - rms) / std(weight))
print "R^2 value: %.2f" % r2_score(weight, predictions)

RME Using model: 16.9
RME Without model: 19.5
Percentage improvement: 13%
R^2 value: 0.26


## Interpretation

Our model came up with the equation `weight = 9.84 * e^(.0122 * height)`. This means for ever 7.8 centimeters height increases, weight increases 10%. If we know how tall someone is, we can predict their weight with an additional accuracy of 13%

## Clarifying Questions

Use this space to ask questions regarding the content covered in the reading. These questions should be restricted to helping you better understand the material. For questions that push beyond what is in the reading, use the next answer field. If you don't have a fully formed question, but are generally having a difficult time with a topic, you can indicate that here as well.

## Enrichment Questions

Use this space to ask any questions that go beyond (but are related to) the material presented in this reading. Perhaps there is a particular topic you'd like to see covered in more depth. Perhaps you'd like to know how to use a library in a way that wasn't show in the reading. One way to think about this is what additional topics would you want covered in the next class (or addressed in a followup e-mail to the class). I'm a little fuzzy on what stuff will likely go here, so we'll see how things evolve.

## Additional Resources / Explorations

If you found any useful resources, or tried some useful exercises that you'd like to report please do so here. Let us know what you did, what you learned, and how others can replicate it.